app/models/proposal.rb

```diff
@@ -69,6 +70,40 @@ class Proposal < ActiveRecord::Base
     where(created_at: start_time...end_time)
   }
 
+  # elasticsearch indexing setup
+  DEFAULT_INDEXED = {include: {client_data: {}, comments: {}, steps: {}}}
+
+  settings index: {
+    number_of_shards: 1, # increase this if we ever get more than N records
+    number_of_replicas: 1
+  } do
+    # with dynamic mapping==true, we only need to explicitly define overrides.
```

Comment:
It is important. I have tweaked it to include an ES doc reference. My fear with putting the full explanation in an external document is that a developer would read the documentation last.